framework/src/modules/interoperability/mainchain/commands/cc_update.ts

```diff
@@ -12,18 +12,433 @@
+		);
+
+		if (!decodedCertificate.validatorsHash.equals(validatorsHash)) {
+			throw new Error('Validators hash is incorrect given in the certificate.');
+		}
+
+		// CCM execution
+		const interoperabilityStore = this.getInteroperabilityStore(context.getStore);
+		const decodedCCMs = txParams.inboxUpdate.crossChainMessages.map(ccm => ({
+			serialized: ccm,
+			deserilized: codec.decode<CCMsg>(ccmSchema, ccm),
+		}));
+		if (partnerChainAccount.status === CHAIN_REGISTERED) {
+			// If the first CCM in inboxUpdate is a registration CCM
+			if (
+				decodedCCMs[0].deserilized.crossChainCommandID === CROSS_CHAIN_COMMAND_ID_REGISTRATION &&
+				decodedCCMs[0].deserilized.receivingChainID === MAINCHAIN_ID
+			) {
+				partnerChainAccount.status = CHAIN_ACTIVE;
+			} else {
+				const beforeSendContext = createBeforeSendCCMsgAPIContext({
+					feeAddress: context.transaction.senderAddress,
+					eventQueue: context.eventQueue,
+					getAPIContext: context.getAPIContext,
+					logger: context.logger,
+					networkIdentifier: context.networkIdentifier,
+				});
+				await interoperabilityStore.terminateChainInternal(
+					txParams.sendingChainID,
+					beforeSendContext,
+				);
+			}
+		}
+
+		for (const ccm of decodedCCMs) {
+			const beforeSendContext = createCCMsgBeforeSendContext({
+				feeAddress: context.transaction.senderAddress,
+				eventQueue: context.eventQueue,
+				getAPIContext: context.getAPIContext,
+				logger: context.logger,
+				networkIdentifier: context.networkIdentifier,
+				getStore: context.getStore,
+			});
+			if (txParams.sendingChainID !== ccm.deserilized.sendingChainID) {
+				await interoperabilityStore.terminateChainInternal(
+					txParams.sendingChainID,
+					beforeSendContext,
+				);
+			}
+			try {
+				validateFormat(ccm.deserilized);
+			} catch (error) {
+				await interoperabilityStore.terminateChainInternal(
+					txParams.sendingChainID,
+					beforeSendContext,
+				);
+			}
+			await interoperabilityStore.appendToInboxTree(
+				getIDAsKeyForStore(txParams.sendingChainID),
+				ccm.serialized,
+			);
+
+			if (ccm.deserilized.receivingChainID !== MAINCHAIN_ID) {
+				await interoperabilityStore.forward({
+					ccm: ccm.deserilized,
+					ccu: txParams,
+					eventQueue: context.eventQueue,
+					feeAddress: context.transaction.senderAddress,
+					getAPIContext: context.getAPIContext,
+					getStore: context.getStore,
+					logger: context.logger,
+					networkIdentifier: context.networkIdentifier,
+				});
+			} else {
+				await interoperabilityStore.apply(
+					{
+						ccm: ccm.deserilized,
+						ccu: txParams,
+						eventQueue: context.eventQueue,
+						feeAddress: context.transaction.senderAddress,
+						getAPIContext: context.getAPIContext,
+						getStore: context.getStore,
+						logger: context.logger,
+						networkIdentifier: context.networkIdentifier,
+					},
+					this.ccCommands,
+				);
+			}
+		}
+
+		// Common ccm execution logic
+		await this._bftAPI.setBFTParameters(
```

Comment:
I think interoperability shouldn't set the BFT params. it should call [updateActiveValidators](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0053.md#update-active-validators)

Reply:
The LIP says `partnerValidators.activeValidators = updateActiveValidators(partnerValidators.activeValidators, activeValidatorsUpdate)` I'm not sure where we set `partnerValidators.activeValidators`, [updateActiveValidators](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0053.md#update-active-validators) is already called above but it doesn't change anything, it simply returns a list but here it asks to set it in partners

Reply:
got confused, we have to use chain validator's store which was completely missing in the code